src/classes/RD2_OpportunityService_TEST.cls

```diff
@@ -199,6 +199,33 @@ private with sharing class RD2_OpportunityService_TEST {
         testAndAssertOppService(rd, expectedOppName);
     }
 
+    /***
+    * @description Verifies the installment opportunities are created with the record types set in the NPSP setting
+    */
+    @isTest
+    private static void oppRecordTypeShouldMatchWithSetting() {
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+
+        Id defaultRecordTypeId = UTIL_RecordTypes.getRecordTypeIdForGiftsTests(Opportunity.sObjectType);
+        UTIL_CustomSettingsFacade.getRecurringDonationsSettingsForTest(new npe03__Recurring_Donations_Settings__c(
+            npe03__Add_Campaign_to_All_Opportunites__c = true,
+            IsRecurringDonations2Enabled__c = true,
+            npe03__Open_Opportunity_Behavior__c = RD2_Constants.CloseActions.Mark_Opportunities_Closed_Lost.name(),
+            npe03__Record_Type__c = defaultRecordTypeId
+        ));
+
+        Test.startTest();
+        createRecurringDonations(10);
+        Test.stopTest();
+
+        List<Opportunity> installmentOpps = getOpportunities();
+        System.assert(!installmentOpps.isEmpty(), 'Opportunities should be created');
```

Comment:
Can we assert expected number of Opportunities to be created instead of verifying we have at least one Opp?